core/src/test/scala/org/apache/spark/sql/delta/LogStoreSuite.scala

```diff
@@ -185,7 +195,7 @@ abstract class LogStoreSuiteBase extends QueryTest
   test("readAsIterator should be lazy") {
     withTempDir { tempDir =>
       val store = createLogStore(spark)
-      val testFile = new File(tempDir, "readAsIterator").getCanonicalPath
+      val testFile = new File(tempDir, "00000000000000000000.json").getCanonicalPath
```

Comment:
why do we need this change?

Reply:
Without this change this test fails (in ExternalLogStoreSuite) with following trace:
```
 - readAsIterator should be lazy *** FAILED *** (7 milliseconds)
[info]   java.lang.AssertionError: assertion failed
[info]   at scala.Predef$.assert(Predef.scala:208)
[info]   at io.delta.storage.BaseExternalLogStore.write(BaseExternalLogStore.scala:186)
```

As you see it fails in BaseExternalLogStore.write - we are parsing filename there to compute `N-1`

I can remove `assert(FileNames.isDeltaFile(path))` from there (should I?),  but it would fail on ` val version = FileNames.deltaVersion(path)`

Reply:
I commented above on the `write` method. Perming an `if (FileNames.isDeltaFile(path))` check should resolve this.

Reply:
It did, thanks!